src/Coordination/KeeperStorage.h

```diff
@@ -35,40 +33,129 @@ class KeeperStorage
     /// New fields should be added to the struct only if it's really necessary
     struct Node
     {
-        /// to reduce size of the Node struct we use a custom Stat without dataLength
-        struct Stat
+        int64_t czxid{0};
+        int64_t mzxid{0};
+        int64_t pzxid{0};
+        uint64_t acl_id = 0; /// 0 -- no ACL by default
+
+        int64_t mtime{0};
+
+        struct
```

Comment:
Consider making these (union/struct) private if you want to avoid using them directly by mistake from other parts of the code.